hammer/tech/__init__.py

```diff
@@ -906,17 +901,21 @@ def filter_for_supplies(self, lib: Library) -> bool:
         :param lib: Library to check
         :return: True if the supplies of this library match the inputs for this run, False otherwise.
         """
-        if lib.supplies is None:
-            # TODO: add some sort of wildcard value for supplies for libraries which _actually_ should
-            # always be used.
-            self.logger.warning("Lib %s has no supplies annotation! Using anyway." % (lib.json()))
-            return True
         # If we are using MMMC assume all libraries will be used.
         # TODO: Read the corners and filter out libraries that don't match any of them.
         # Requires a refactor because MMMCCorner parsing is only in HammerTool now.
         # See issue #275.
         if self.get_setting("vlsi.inputs.mmmc_corners"):
             return True
+        if lib.supplies is None:
+            # TODO: add some sort of wildcard value for supplies for libraries which _actually_ should
+            # always be used.
+            if lib.provides is not None:
+                for provided in lib.provides:
+                    if provided.lib_type is not None and provided.lib_type == "technology":
+                        return True
```

Comment:
Mostly. This is because the technology provides is usually only the tech LEF where we don't have the supplies. There might be other libraries that are supplies-agnostic that could still throw errors but we'd need to enforce categories of provides in order to filter them out.